erdos/src/dataflow/connect.rs

```diff
@@ -16,14 +16,12 @@ use crate::{
 
 /// Adds a [`Source`] operator, which has no read streams, but introduces data into the dataflow
 /// graph by interacting with external data sources (e.g., other systems, sensor data).
-pub fn connect_source<O, S, T>(
+pub fn connect_source<O, T>(
     operator_fn: impl Fn() -> O + Clone + Send + Sync + 'static,
-    state_fn: impl Fn() -> S + Clone + Send + Sync + 'static,
     mut config: OperatorConfig,
 ) -> Stream<T>
 where
-    O: 'static + Source<S, T>,
-    S: State,
```

Comment:
Are there cases where we'd want to provide a system-managed state for sources? I'm thinking in case of system-managed checkpointing for fault tolerance, or operator migration.

In any case, I'm happy to make this change, and potentially re-introduce state for other purposes in a future release.

Reply:
It's not immediately clear to me how to provide *automatic* checkpointing for `Source` operators since the `run` method is supposed to be a place where the developer takes control. They could still use our `TimeVersionedState` but they would have to call their own commits and stuff.

My current take is to remove it, and if we come up with a clean way to do automatic checkpointing, we can potentially re-introduce it.

Reply:
Could you update the documentation in `lib.rs` to remove the state initialization?